src/sideBar.ts

```diff
@@ -0,0 +1,737 @@
+                };
+                this.tooltip = 'Debug';
+                this.collapsibleState = TreeItemCollapsibleState.Expanded;
+                this.contextValue = 'debug';
+                break;
+            case TargetType.Launch:
+                this.command = {
+                    title: localize('Launch', 'Launch'),
+                    command: 'cmake.sideBar.launchTarget',
+                    arguments: [BaseNode.cmakeProject.workspaceFolder, BaseNode.cmakeProject.launchTargetName || await BaseNode.cmakeProject.allTargetName]
+                };
+                this.tooltip = 'Launch';
+                this.collapsibleState = TreeItemCollapsibleState.Expanded;
+                this.contextValue = 'launch';
+                break;
+            case TargetType.Project:
+                this.tooltip = 'Active project';
+                this.collapsibleState = TreeItemCollapsibleState.Expanded;
+                this.contextValue = 'activeProject';
+                break;
+        }
+
+        await this.InitializeChildren();
+    }
+
+    async InitializeChildren(): Promise<void> {
+        if (!BaseNode.cmakeProject) {
+            return;
+        }
+        switch (this.targetType) {
+            case TargetType.Debug:
+                this.target = new TargetNode(TargetType.Debug);
+                break;
+            case TargetType.Launch:
+                this.target = new TargetNode(TargetType.Launch);
+                break;
+            case TargetType.Project:
+                this.target = new TargetNode(TargetType.Project);
+                break;
+            default:
+                throw Error('Not a valid type');
+        }
+        await this.target!.initialize();
+    }
+
+    getChildren(): BaseNode[] {
+        if (!BaseNode.cmakeProject) {
+            return [];
+        }
+        return [this.target!];
+    }
+
+}
+
+export class PresetNode extends BaseNode {
+
+    constructor(private presetType: PresetType) {
+        super(NodeType.Kit);
+    }
+
+    async initialize(): Promise<void> {
+        if (!BaseNode.cmakeProject || !BaseNode.cmakeProject.useCMakePresets) {
+            return;
+        }
+        switch (this.presetType) {
+            case PresetType.Configure:
+                this.label = BaseNode.cmakeProject.configurePreset?.name || noConfigPresetSelected;
+                this.command = {
+                    title: localize('change.preset', 'Change Preset'),
+                    command: 'cmake.sideBar.selectConfigurePreset',
+                    arguments: [this]
+                };
+                this.tooltip = 'Change Configure Preset';
+                this.contextValue = 'configPreset';
+                break;
+            case PresetType.Build:
+                this.label = BaseNode.cmakeProject.buildPreset?.name || noBuildPresetSelected;
```

Comment:
We'll want to have a friendly name for the "default" build/test presets. In the GIF, it looks like the internal name for the test preset is showing.

Reply:
I changed it to [default]